File: android/src/main/java/com/yugioh/android/base/BaseActivity.java
package com.yugioh.android.base;

import android.content.res.Configuration;
import com.yugioh.android.R;

public abstract class BaseActivity extends InnerActivity {

    @Override
    public boolean getCondition() {
        return getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
    }

    @Override
    public int getBaseLayout() {
        return R.layout.layout_replacement;
    }

    @Override
    public int getReplaceId() {
        return R.id.fReplacement;
    }

}

File: android/src/main/kotlin/com/yugioh/android/base/BaseActivity.kt
package com.yugioh.android.base

import android.content.res.Configuration
import com.yugioh.android.R

/**
 * Created by rarnu on 3/23/16.
 */
abstract class BaseActivity: InnerActivity() {

    override fun getCloseCondition(): Boolean = resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE

    override fun getBaseLayout(): Int = R.layout.layout_replacement

    override fun getReplaceId(): Int = R.id.fReplacement

}